zucchini/graders/pylc3_grader.py

```
@@ -0,0 +1,92 @@
+    """
+
+    DEFAULT_TIMEOUT = 30
+
+    def __init__(self, test_class, timeout=None):
+        self.test_class = test_class
+
+        if timeout is None:
+            self.timeout = self.DEFAULT_TIMEOUT
+        else:
+            self.timeout = timeout
+
+    def list_prerequisites(self):
+        return ['build-essential', 'g++', 'cmake', 'libboost-all-dev',
+                'libglib2.0-dev', 'castxml', 'python-pip']
+
+    def list_extra_setup_commands(self):
+        return ['pip install scikit-build',
+                'pip install pyLC3',
+                'ldconfig',
+                'pip install parameterized']
+
+    def part_from_config_dict(self, config_dict):
+        return PyLC3Test.from_config_dict(config_dict)
+
+    def grade(self, submission, path, parts):
+        cmdline = ['python2', 'runner.py', self.test_class]
+        env = os.environ.copy()
+        # HACK HACK HACK HACK HACK
+        env['LD_LIBRARY_PATH'] = '/usr/local/lib/python2.7/dist-packages/pyLC3'
+        try:
+            # Do not mix stderr into stdout because sometimes our friend
+            # Roi printStackTrace()s or System.err.println()s, and that
+            # will mess up JSON parsing
+            process = run_process(cmdline, cwd=path, timeout=self.timeout,
+                                  stdout=PIPE, stderr=STDOUT, input='',
+                                  env=env)
+        except TimeoutExpired:
+            raise BrokenSubmissionError('timeout of {} seconds expired for '
+                                        'grader'.format(self.timeout))
+
+        if process.returncode != 0:
+            raise BrokenSubmissionError(
+                'grader command exited with nonzero exit code {}'
+                .format(process.returncode),
+                verbose=(process.stdout.decode()
+                         if process.stdout else '(no output)'))
+
+        with open(os.path.join(path, 'results.json'), 'r') as json_fp:
+            results = json.load(json_fp)
```

need to throw a better error if the tester fails to produce the json